src/foam/nanos/approval/ApprovableAware.js

```diff
@@ -74,19 +75,46 @@
                 diff = new TreeMap<>();
                 diff.putAll(diffHashmap);
               }
-
+              
               StringBuilder hash_sb = new StringBuilder(obj.getClass().getSimpleName());
               if ( operation == Operations.UPDATE && obj instanceof ApprovableAware ) 
                 hash_sb.append(String.valueOf(obj.getProperty("id")));
 
               if ( diff != null ) {
-                // remove ids, timestamps and userfeedback
+                // remove ids, timestamps and storageTransient properties
                 if ( operation == Operations.CREATE ) diff.remove("id");
                 diff.remove("created");
                 diff.remove("lastModified");
-                diff.remove("userFeedback");
 
-                // convert array properties to list to get consistent hash
+                // remove lifecycle state
+                diff.remove("lifecycleState");
+
+                Iterator allProperties = obj.getClassInfo().getAxiomsByClass(PropertyInfo.class).iterator();
+
+                List<String> storageTransientPropertyNames = new ArrayList<>();
+                List<String> networkTransientPropertyNames = new ArrayList<>();
+        
+                while ( allProperties.hasNext() ){
+                  PropertyInfo prop = (PropertyInfo) allProperties.next();
+        
+                  if ( prop.getStorageTransient() ){
+                    storageTransientPropertyNames.add(prop.getName());
+                  }
+
+                  if ( prop.getNetworkTransient() ){
```

Comment:
shouldn't we be keeping network transient properties? these properties are stored in the database and would contribute to a valid change in the object, no?

Reply:
This is for when generating the lookupKey as the transient properties would interfere with the lookup.

However on the approvableAwareDAO I remove only storageTransient properties during updates, as when I'm updating lets say just a name on user, when viewing the reference for what properties were updated I would see userFeedback and on account I would see balance (local) and balance (home)

What we can do then is instead is:
1. HIDE storage transient and network transient properties in the PropertiesToUpdateView.
2. Keep the removal of storage transient and network transient properties when generating the lookup hash key, but have both storage transient and network transient properties present when generating the maps which get used to create the final object put in the DAO.

@nanoscott @jlhughes

Reply:
1. OK
2. OK.

Reply:
I'm good with it too.

Reply:
1. and 2. just added.